integration/use-cases/create-private-registry.js

```diff
@@ -0,0 +1,96 @@
+const axios = require("axios");
+
+test("Creates a private registry", async () => {
+  await page.goto(getUrl("/#/config/ns/default/repos"));
+
+  await expect(page).toFillForm("form", {
+    token: process.env.ADMIN_TOKEN,
+  });
+
+  await expect(page).toClick("button", { text: "Login" });
+
+  await expect(page).toClick("button", { text: "Add App Repository" });
+
+  try {
+    await expect(page).toMatch("Install Repo");
+  } catch (e) {
+    // The Modal sometimes fail to be opened, click the button again
+    await expect(page).toClick("button", { text: "Add App Repository" });
+
+    await expect(page).toMatch("Install Repo");
+  }
+
+  await page.type("#kubeapps-repo-name", "my-repo");
+
+  await page.type(
+    "#kubeapps-repo-url",
+    "http://chartmuseum-chartmuseum.kubeapps:8080"
+  );
+
+  await expect(page).toClick("label", { text: "Basic Auth" });
+
+  // Credentials from e2e-test.sh
+  await page.type("#kubeapps-repo-username", "admin");
+  await page.type("#kubeapps-repo-password", "password");
+
+  // Open form to create a new secret
+  const randomNumber = Math.floor(Math.random() * Math.floor(100));
+  const secret = "my-repo-secret" + randomNumber;
+  await expect(page).toClick("a", { text: "Add new credentials" });
+  await page.type("#kubeapps-docker-cred-secret-name", secret);
+  await page.type(
+    "#kubeapps-docker-cred-server",
+    "https://index.docker.io/v1/"
+  );
+  await page.type("#kubeapps-docker-cred-username", "user");
+  await page.type("#kubeapps-docker-cred-password", "password");
+  await page.type("#kubeapps-docker-cred-email", "user@example.com");
+  await expect(page).toClick("button", { text: "Submit" });
+
+  // Select the new secret
+  await expect(page).toClick("label", { text: secret });
```

Comment:
I meant to ask about this: in the UX we should automatically select the secret which was just created, don't you think? (It'd be easy to create it and not select it).

Reply:
it's not exactly trivial to do this since the component just re-renders with a new set of secrets in the props but we can take that separately.